src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Diagnostics.DataProviders.Interfaces;
+using Newtonsoft.Json;
+using Diagnostics.ModelsAndUtils.Models.ChangeAnalysis;
+using Diagnostics.DataProviders.TokenService;
+using System.Collections.Generic;
+using System.Net;
+
+namespace Diagnostics.DataProviders
+{
+    public class ChangeAnalysisClient : IChangeAnalysisClient
+    {
+        /// <summary>
+        /// x-ms-client-object-id header to pass to Change Analysis endpoint.
+        /// </summary>
+        private string clientObjectIdHeader;
+
+        /// <summary>
+        /// For detectors loaded from Diagnose and Solve, pass x-ms-client-principal-name to Change Analysis endpoint.
+        /// </summary>
+        private string clientPrincipalNameHeader;
+
+        private readonly string changeAnalysisEndPoint = "https://changeanalysis-dataplane-dev.azurewebsites.net/providers/microsoft.changeanalysis/";
```

Comment:
The pattern we adopted for this project is load values like these as configuration values from DataProviderConfigurations. Add another property for endpoint in `ChangeAnalysisDataProviderConfiguration` and that value will be loaded from registry in production or the appsettings when run in other environments.

Reply:
Sure thats a valid point.